usr/src/uts/common/fs/zfs/zvol.c

```diff
@@ -1811,26 +1817,21 @@ zvol_ioctl(dev_t dev, int cmd, intptr_t arg, int flag, cred_t *cr, int *rvalp)
 
 		zfs_range_unlock(rl);
 
-		if (error == 0) {
```

Comment:
Good question, I had to go back and read the comments in our code review for this. We were fixing a closely related bug -- the `txg_wait_synced()` wasn't necessary. For more details you can check out these comments from the mailing list:

- https://www.listbox.com/member/archive/274414/2016/05/sort/time_rev/page/2/entry/6:34/20160502161605:B03E3D66-10A2-11E6-B4E7-EC1002035550/
- https://www.listbox.com/member/archive/274414/2016/05/sort/time_rev/page/2/entry/5:34/20160502163156:E72188F4-10A4-11E6-9FA9-FC9C7BBAF48C/